src/traccuracy/metrics/_base.py

```diff
@@ -21,13 +22,95 @@ class Metric(ABC):
     supports_many_to_many = False
 
     @abstractmethod
-    def compute(self, matched: Matched) -> dict:
+    def _compute(self, matched: Matched) -> dict:
         """The compute methods of Metric objects return a dictionary with counts and statistics.
 
+        Args:
+            matched (traccuracy.matchers.Matched): Matched data object to compute metrics on
+
         Raises:
             NotImplementedError
 
         Returns:
             dict: Dictionary of metric names and int/float values
         """
         raise NotImplementedError
+
+    def compute(self, matched: Matched) -> Results:
+        """The compute methods of Metric objects returns a Results object populated with results
+        and associated metadata
+
+        Args:
+            matched (traccuracy.matchers.Matched): Matched data object to compute metrics on
+
+        Returns:
+            Results: Object containing metric results and associated pipeline metadata
+        """
+
+        res_dict = self._compute(matched)
+
+        results = Results(
+            results=res_dict,
+            matcher_info=matched.matcher_info,
+            metric_info=self.info,
+            gt_name=matched.gt_graph.name,
+            pred_name=matched.pred_graph.name,
+        )
+        return results
+
+    @property
+    def info(self):
+        """Dictionary with Matcher name and any parameters"""
+        return {"name": self.__class__.__name__, **self.__dict__}
+
+
+class Results:
+    """The Results object collects information about the pipeline used
+    to generate the metric results
+
+    Args:
+        results (optional, dict): Dictionary with metric output
+        matcher_info (optional, dict): Dictionary with matcher name and parameters
+        metric_info (optional, dict): Dictionary with metric name and parameters
+        gt_name (optional, str): Name of the ground truth data
+        pred_name (optional, str): Name of the predicted data
+    """
+
+    def __init__(
+        self,
+        results: dict | None = None,
+        matcher_info: dict | None = None,
```

Comment:
Results, matcher_info, and metric_info can be required and always included in the to_dict